server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -43,12 +51,39 @@ public class DesiredBalanceComputer {
 
     private static final Logger logger = LogManager.getLogger(DesiredBalanceComputer.class);
 
+    private final Settings settings;
+    private final ClusterSettings clusterSettings;
+    private final ThreadPool threadPool;
     private final ShardsAllocator delegateAllocator;
 
     protected final MeanMetric iterations = new MeanMetric();
 
-    public DesiredBalanceComputer(ShardsAllocator delegateAllocator) {
+    public static final Setting<TimeValue> PROGRESS_LOG_INTERVAL_SETTING = Setting.timeSetting(
+        "cluster.routing.allocation.desired_balance.progress_log_interval",
+        TimeValue.timeValueMinutes(1),
+        TimeValue.timeValueSeconds(1),
```

Comment:
I don't think we need a particularly restrictive minimum on this setting. Zero would log every iteration which might be a little silly but not totally unreasonable?